src/main/java/bio/terra/workspace/service/crl/CrlService.java

```diff
@@ -0,0 +1,116 @@
+package bio.terra.workspace.service.crl;
+
+import bio.terra.cloudres.common.ClientConfig;
+import bio.terra.cloudres.common.cleanup.CleanupConfig;
+import bio.terra.cloudres.google.billing.CloudBillingClientCow;
+import bio.terra.cloudres.google.cloudresourcemanager.CloudResourceManagerCow;
+import bio.terra.cloudres.google.serviceusage.ServiceUsageCow;
+import bio.terra.workspace.app.configuration.external.CrlConfiguration;
+import bio.terra.workspace.service.crl.exception.CrlInternalException;
+import bio.terra.workspace.service.crl.exception.CrlNotInUseException;
+import bio.terra.workspace.service.crl.exception.CrlSecurityException;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.time.Duration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CrlService {
+  /** The client name required by CRL. */
+  public static final String CLIENT_NAME = "workspace";
+  /** How long to keep the resource before Janitor do the cleanup. */
+  public static final Duration TEST_RESOURCE_TIME_TO_LIVE = Duration.ofHours(1);
+
+  private final CrlConfiguration crlConfig;
+
+  private final CloudResourceManagerCow crlResourceManagerCow;
+  private final CloudBillingClientCow crlBillingClientCow;
+  private final ServiceUsageCow crlServiceUsageCow;
+
+  @Autowired
+  public CrlService(CrlConfiguration crlConfig) {
+    this.crlConfig = crlConfig;
+
+    ClientConfig clientConfig;
+    if (crlConfig.useCrl()) {
+      var creds = buildCredentials();
```

Comment:
Do we have guidelines on when to use `var`? My personal preference, fwiw, would be to use it when the type is obvious entirely e.g. `var x = new Foo();` or long, uninteresting, and somewhat obvious. I don't think the return value of a member function is either, and there are many possible "credentials" types in general.

Reply:
We don't, but point taken. I agree if it is not obvious what the type is without looking elsewhere, then you shouldn't use var. Will fix.